cognite/client/data_classes/data_modeling/queries.py

```diff
@@ -0,0 +1,343 @@
+from __future__ import annotations
+
+import json
+from abc import ABC, abstractmethod
+from collections import UserDict
+from dataclasses import dataclass, field
+from typing import Any, Dict, Literal, Mapping, Optional, Type
+
+from cognite.client.data_classes.data_modeling.filters import Filter
+from cognite.client.data_classes.data_modeling.ids import ViewId
+from cognite.client.data_classes.data_modeling.instances import (
+    Edge,
+    EdgeList,
+    InstanceSort,
+    Node,
+    NodeList,
+    PropertyValue,
+)
+
+
+@dataclass
+class SourceSelector:
+    source: ViewId
+    properties: list[str]
+
+    def dump(self, camel_case: bool = False) -> dict[str, Any]:
+        return {
+            "source": self.source.dump(camel_case),
+            "properties": self.properties,
+        }
+
+    @classmethod
+    def load(cls, data: dict | str) -> SourceSelector:
+        data = json.loads(data) if isinstance(data, str) else data
+        return cls(
+            source=ViewId.load(data["source"]),
+            properties=data["properties"],
+        )
+
+
+@dataclass
+class Select:
+    sources: list[SourceSelector] = field(default_factory=lambda: [])
+    sort: list[InstanceSort] = field(default_factory=lambda: [])
+    limit: Optional[int] = None
+
+    def dump(self, camel_case: bool = False) -> dict[str, Any]:
+        output: Dict[str, Any] = {}
+        if self.sources:
+            output["sources"] = [
+                {"source": source.source.dump(camel_case), "properties": source.properties} for source in self.sources
+            ]
+        if self.sort:
+            output["sort"] = [s.dump(camel_case) for s in self.sort]
+        if self.limit:
+            output["limit"] = self.limit
+        return output
+
+    @classmethod
+    def load(cls, data: dict | str) -> Select:
+        data = json.loads(data) if isinstance(data, str) else data
+        return cls(
+            sources=[SourceSelector.load(source) for source in data.get("sources", [])],
+            sort=[InstanceSort._load(s) for s in data.get("sort", [])],
+            limit=data.get("limit"),
+        )
+
+
+class Query:
```

Comment:
What would be very nice is a `load()` method which accepts a yaml (or json as that is valid yaml) string. I often want to just quickly write up a query using yaml.